### PR TITLE
added calcDim Proc for Entity

### PR DESCRIPTION
**Pull request by CodeDoes**

### CODE REVIEW

File: nimgame2/entity.nim [hunk 1]
```diff
@@ -700,7 +700,12 @@ proc absPos*(entity: Entity): Coord =
                entity.parent.absPos * entity.parallax,
                entity.absRot)
 
-
+proc calcDim(entity: Entity): Dim=
```
Comment (Vladar4) [inline nimgame2/entity.nim]: * Rename to `dim`
* change into the template
* Make public

**Review by Vladar4 — CHANGES_REQUESTED** [review]
Might be useful but need some improvements:

**Comment by CodeDoes** [comment]
Yes. this is a tentative approach. I've got a lot of other code I want to share. Just wanted your thoughts first.
I have a "Transform" structure.
I have relative_point, inverse_point procs. I see you already have something similar in colliders.nim.
I wrote a method to get the "topleft" "bottomleft" "topright" "bottomright" relative (local, without scale and rotation point) and inverse points (global)

**Comment by Vladar4** [comment]
Thanks!

> I've got a lot of other code I want to share. Just wanted your thoughts first.

I'd prefer to review such additions in separate pull requests.

**Comment by CodeDoes** [comment]
I'll make a pull request for your review then.

**Pull request merged**